rule/add_constant.go

```diff
@@ -36,13 +38,16 @@ type AddConstantRule struct {
 	allowList       allowList
 	ignoreFunctions []*regexp.Regexp
 	strLitLimit     int
-
-	configureOnce sync.Once
+	configureOnce   sync.Once
 }
 
 // Apply applies the rule to given file.
-func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
-	r.configureOnce.Do(func() { r.configure(arguments) })
+func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) ([]lint.Failure, error) {
+	var configureErr error
+	r.configureOnce.Do(func() { configureErr = r.configure(arguments) })
+	if configureErr != nil {
+		return nil, configureErr
+	}
```

Comment:
If `r.configure` fails, only the first call will receive the error. Any subsequent calls—second, third, and so on—won’t return that error.

See this article https://victoriametrics.com/blog/go-sync-once/index.html

Maybe we can fix this by adding the `configureErr error` field to the `AddConstantRule` struct.

Reply:
Yes i know that, and we want to revive stops when first error exists, by the way  ```configureErr error``` field was replaced by local var few commits ago :P

Reply:
I suggested using sync.OnceFunc earlier

https://github.com/mgechev/revive/pull/1126#discussion_r1845293327

I'm unsure if it was understood, or if I'm the one who misundertood how it works

What about https://pkg.go.dev/sync#OnceValue then ?

Reply:
ok @ccoVeille, @alexandear  that code works:

```
check := sync.OnceValue(func() error { return r.configure(arguments) })                                                                                                
 if err := check(); err != nil {                                                                                                                                        
 return nil, err                                                                                                                                                    
 }
```
but what about with ```configureOnce sync.Once``` in rule struct, because if we decide to above code, then that var is useless, and if that solution is the final one because modifing 34+ files every secound commit is not optimal workflow :P